src/time_tagger/connection/service.py

```diff
@@ -0,0 +1,110 @@
+from dataclasses import dataclass
+from shared.messages.error_messages_enum import ErrorMessageEnum
+from time_tagger.connection.dao import ConnectionDao
+
+from time_tagger.connection.repository import (
+    ConnectionStatusEnum,
+    ConnectionRepository,
+    TimeTaggerConnectionInfoDto,
+)
+from time_tagger.hardware_properties.dao import TimeTaggerHardwarePropertiesDao
+
+
+@dataclass
+class TimeTaggerAddressInfo:
+    host_address: str
+    port: str
+    time_tagger_name: str
+
+
+@dataclass
+class TimeTaggerConnectionRes:
+    connected_devices: list[TimeTaggerConnectionInfoDto]
+    connection_failed_devices: list[TimeTaggerConnectionInfoDto]
+
+
+class ConnectionService:
+    def __init__(
+        self,
+        time_tagger_connection_data: ConnectionRepository,
+        time_tagger_hardware_service: TimeTaggerHardwarePropertiesDao,
+        connection_dao: ConnectionDao,
+    ):
+        self.connection_data = time_tagger_connection_data
+        self.time_tagger_hardware_service = time_tagger_hardware_service
+        self.connection_dao = connection_dao
+
+    def connect_to_time_tagger_server(
+        self,
+        time_taggers_info: list[TimeTaggerAddressInfo],
+    ) -> TimeTaggerConnectionRes:
+
+        connected_devices = []
+        connection_failed_devices = []
+
+        for time_tagger in time_taggers_info:
+            try:
```

Comment:
the try scope is getting bigger. 
It's out of the scope of this PR, but we should consider moving it to a function for readability